### PR TITLE
ui: add settings view

### PR DESCRIPTION
**Pull request by leonardojoao**

### CODE REVIEW

**Comment by gustavosbarreto** [comment]
See #215

File: ui/src/router/index.js [hunk 1]
```diff
@@ -64,6 +64,21 @@ const router = new Router({
       name: 'firewalls',
       component: () => import('./../views/FirewallRules.vue'),
     },
+    {
+      path: '/settings',
+      name: 'settings',
+      component: () => import(/* webpackChunkName: 'devices' */ './../views/Settings.vue'),
```
Comment (leonardojoao) [inline ui/src/router/index.js:70]: webpackChunkName: 'devices' or /* webpackChunkName: 'settings'?

File: api/user/service.go [hunk 2]
```diff
@@ -24,12 +24,8 @@ func NewService(store store.Store) Service {
...
 	user, _ := s.store.GetUserByTenant(ctx, tenant)
 	if user != nil {
-			if username == "" && email == "" { //change in password
-				return s.store.UpdateUser(ctx, user.Username, user.Email, password, tenant)
-			} else if password == "" { //changes in the name and email
-			if user.Username != username || user.Email != email {
-				return s.store.UpdateUser(ctx, username, email, user.Password, tenant)
-			}
+		if user.Username != username || user.Email != email {
+			return s.store.UpdateUser(ctx, username, email, user.Password, tenant)
```
Comment (otavio) [inline api/user/service.go:28]: Could it be done like this?

```
func (s *service) UpdateDataUser(ctx context.Context, username, email, password, tenant string) error {
	user, _ := s.store.GetUserByTenant(ctx, tenant)
	if user == nil {
                return ErrUnauthorized
        }
	
        return s.store.UpdateUser(ctx, username, email, user.Password, tenant)
}
```

File: api/store/mongo/store.go [hunk 3]
```diff
@@ -827,6 +827,34 @@ func (s *Store) GetRecord(ctx context.Context, uid models.UID) ([]models.Recorde
...
+
+    if password!="" {
+			if _, err := s.db.Collection("users").UpdateOne(ctx, bson.M{"tenant_id": tenant}, bson.M{"$set": bson.M{"password": password}}); err!= nil {
+					return err
+				}
+			}
+
+    return nil
+}
+
```
Comment (eduardoveiga) [inline api/store/mongo/store.go:857]: By this way all logic of what will be updated is inside store, not services. @gustavosbarreto it's ok?

Comment (eduardoveiga) [inline api/store/mongo/store.go:857]: By this way all logic of what will be updated is inside store, not services. @gustavosbarreto it's ok?

File: ui/src/components/setting/SettingProfile.vue [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+        newPassword: this.newPassword,
+      };
+
+      try {
+        await this.$store.dispatch('users/put', data);
+        this.$store.dispatch('modals/showSnackbarSuccess', true);
+      } catch {
+        this.$store.dispatch('modals/showSnackbarError', true);
+      }
+      this.enableEdit();
```
Comment (leonardojoao) [inline ui/src/components/setting/SettingProfile.vue:299]: I would put in try, because the request fails the dialog not close.

**Pull request merged**